### PR TITLE
Add FindChallengeBottomBar and refactor challenges validation

### LINKED ISSUE #59: Challenge fail feedback
Currently, there is no feedback when attempting a challenge but failing.

As a learner who has attempted to complete a challenge by pasting text but has not passed it successfully, I should receive some feedback and possibly a hint.

### LINKED ISSUE #70: Create a FindChallengeBottomBar component
We have two "Find" challenges ([one](https://savingsatoshi.com/chapters/chapter-1/genesis), [two](https://savingsatoshi.com/chapters/chapter-1/transacting/transacting-1)). The bottom bars function identically for them. Each page implements this, but it can be turned into a component.

The states are default, incorrect, and correct. Text should be customizable per page, and mobile layouts should be supported. On mobile, we need to figure out whether the component should be sticky to the bottom of the page or not (need to feel this out).

We may also be able to use this for the Terminal-style challenges.

**Design reference ([Figma](https://www.figma.com/file/LqjK3Tpvd9KJ4buFArCJBQ/Saving-Satoshi?node-id=1195%3A16656&t=nxCE26gS2fD24CVh-1)):**

image

**In context:**

image

### PR DESCRIPTION
**Pull request by aureleoules**
Fixes #70
Fixes #59

This PR adds a `FindChallengeBottomBar` that now handles the input validation with the props `input` and `expected`. I think it makes the user input validation flow easier.

image
image
image

image
image

### CODE REVIEW

**Comment by GBKS** [comment]
Looks great. A few small things I noticed:

- The success background should apply to the container on mobile, not just to the text
- In non-success states the component background should be 15% black
- The success message should have a small check icon

**Screenshot** 
image

**Design**
image

**Comment by aureleoules** [comment]
Thanks for the review @GBKS, updated.

image

image

File: app/chapters/[slug]/transacting/transacting-2/page.tsx [hunk 1]
```diff
@@ -85,15 +72,19 @@ export default function Genesispt2() {
...
                 <span className="flex h-7 w-7 items-center justify-center rounded bg-white">
                   <PlayIcon className="text-black" />
                 </span>
                 Run the script
               </button>
-              <SignUpModal onConfirm={onSignUp} onClose={() => setOpenModal(false)} open={openModal} />
             </div>
+            )}
+            {success && (
+              <FindChallengeBottomBar next="/" input={"true"} expected={"true"} />
```
Comment (aureleoules) [inline app/chapters/[slug]/transacting/transacting-2/page.tsx:86]: Note that this should lead to `/chapters/chapter-1/done` once #72 is merged.

File: components/chapters/FindChallengeBottomBar.tsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+            max-md:gap-4
+          ">
+          <div
+            className='flex w-full items-center align-middle transition duration-150 ease-in-out md:px-5'
+          >
+            <p
+              className={clsx(
+                'font-nunito text-[21px] text-white transition duration-150 ease-in-out',
+                {
+                  'opacity-50': getStatus() !== Status.Success,
```
Comment (GBKS) [inline components/chapters/FindChallengeBottomBar.tsx:44]: Sorry, small thing I didn't catch in the last feedback. The text should be white in error state also.

**Review by GBKS — APPROVED** [review]
Ack. Looks great. Thanks for addressing my feedback so quickly.

**Pull request merged**